lib/topic_query.rb

```diff
@@ -269,6 +269,14 @@ def list_private_messages_group_archive(user)
     create_list(:private_messages, {}, list)
   end
 
+  def list_private_messages_tag(user)
+    list = private_messages_for(user, :all)
+    tag_id = Tag.where('name ilike ?', @options[:tags][0]).pluck(:id).first
+    list = list.joins("JOIN topic_tags tt ON tt.topic_id = topics.id AND
+                      tt.tag_id = #{tag_id}")
```

Comment:
I'm wondering if we can combine the query ran in `tag_id` into a sub-query here. It'll allow us to send only one query to the DB.